src/ElasticsearchEngine.php

```diff
@@ -11,24 +11,44 @@
 class ElasticsearchEngine extends Engine
 {
     /**
-     * Index where the models will be saved.
+     * Default index where the models will be saved.
      *
      * @var string
      */
     protected $index;
 
+    /**
+     * If the index should be set per model.
+     *
+     * @var bool
+     */
+    protected $perModelIndex;
+
     /**
      * Create a new engine instance.
      *
      * @param  \Elasticsearch\Client  $elastic
      * @return void
      */
-    public function __construct(Elastic $elastic, $index)
+    public function __construct(Elastic $elastic, $index, $perModelIndex = false)
     {
         $this->elastic = $elastic;
         $this->index = $index;
+        $this->perModelIndex = $perModelIndex;
+    }
+
+    /**
+     * Retrieves the index for the given model.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @return string
+     */
+    protected function getIndex($model)
+    {
+        return ($this->perModelIndex ? $model->searchableAs() : $this->index);
```

Comment:
It might be nice to treat `$this->index` as an "index prefix" if `perModelIndex` is set to true. For our use-case, we need the option to keep multiple indexes for the same model, for swapping during development. In our fork, I changed L48 to this:

    return ($this->perModelIndex ? $this->index . $model->searchableAs() : $this->index);

Just something for consideration. This is similar to how @intelligide is handling this issue in #91.